power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/utils.hpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_UTILS_HPP
+#define POWER_GRID_MODEL_CPP_UTILS_HPP
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+#include <algorithm>
+#include <complex>
+#include <limits>
+#include <type_traits>
+
+namespace power_grid_model_cpp {
+template <typename T>
+inline typename std::enable_if<std::is_floating_point<T>::value, // NOLINT(modernize-use-constraints,
+                                                                 // modernize-type-traits)
+                               bool>::type
+is_nan(T x) {
+    return std::isnan(x);
+}
+template <typename T>
+inline typename std::enable_if<std::is_floating_point<T>::value, // NOLINT(modernize-use-constraints,
+                                                                 // modernize-type-traits)
+                               bool>::type
+is_nan(std::complex<T> const& x) {
+    return is_nan(x.real()) || is_nan(x.imag());
+}
+inline bool is_nan(int32_t x) { return x == std::numeric_limits<int32_t>::min(); }
+inline bool is_nan(int8_t x) { return x == std::numeric_limits<int8_t>::min(); }
```

Comment:
please add `Idx` and `ID` instead. @TonyXiang8787 should we also add an additional type `IntS` instead of `int8_t`?

Reply:
interestingly, there is no enum value for 64-bit ctype (`Idx`), only for 32-bit ctype (`ID`). is there any particular reason for that?

Reply:
We do not have `CType` of 64-bit integer in any attributes. `Idx` is not an attribute!

Reply:
So we do not need to add `Idx` in this overload set. It is never used.

Reply:
Define `IntS` is fine.

Reply:
All the `is_nan` overloads were taken from `three_phase_tensor.hpp`, and I removed the ones that I thought were not useful. The `Idx` overload sliped my mind, I will remove and add `IntS` as well.